### ISSUE TITLE
Applets internal endpoint has http hardocded in the url

### ISSUE THREAD

**Issue by matteo-cristino** [ForkbombEu/ncr#172]
This make them fail when they are using from https pages

https://github.com/ForkbombEu/ncr/blob/main/src/index.ts#L435

**Comment by puria** [comment]
The trick here is to leave it without protocol and use relative URLs.

TL;DR is:

> [!TIP]
> ```
>   				endpoint: `//${config.hostname}:${config.port}${path}`
> ```

RTFM is:

A relative URL is a URL that does not include the complete protocol, domain, and port information. Instead, it specifies the path or location relative to the current page or resource. It allows you to reference resources within the same domain or subdomain without explicitly providing all the details of the base URL.

In JavaScript running in a browser environment, when you use a relative URL without specifying a protocol (e.g., `//example.com/resource`), it inherits the protocol (HTTP or HTTPS) of the current webpage. This behavior is defined in HTML5's specification for URLs known as RFC 3986.

By omitting the protocol from a URL, you ensure that your code will work correctly regardless of whether your website uses HTTP or HTTPS protocols. The browser automatically replaces `//` with either `http://` or `https://`, based on how your page was loaded.

It's worth noting that this behavior only applies when working with URLs inside web browsers. In Node.js environments, where there is no inherent concept of browsing webpages, using relative URLs without specifying protocols may result in errors unless handled differently using additional modules like `url-parse`.
